### DISCUSSION TITLE
Add conductance parameter to PstFrom object

### DISCUSSION THREAD

**Discussion by tamvejarf** [pypest/pyemu#553] [Q&A]
I have a MODFLOW-USG model created with Groundwater Vistas 8, and I am using PyEMU to generate a .pst file. I’ve created a PstFrom object and need to add a parameter for river conductance. My parameters will include hydraulic properties, river and drain conductance, and pumping rates. The structure of the package is shown in the attached image.

I would like to know if it is possible to use the add_parameters function to specify that the conductance should be parameterized and grouped by its reach (based on the 3rd and 6th columns in the file). Additionally, can this function generate the required .tpl files, or is it necessary to define external .tpl files manually and build the .pst file from those?

![riv](https://github.com/user-attachments/assets/e5b0fc58-e98f-4a6e-a481-0b8358f2104c)

**Comment by briochh** [comment]
It definitely should be possible to parameterise this package using PstFrom.add_parameters(). It will be easiest if you can store the stress period data to individual external files. Then you just pass index_cols (probably 0) and use_cols (whichever col is conductance).

**Reply by briochh** [reply]
You may need to get creative with your index_cols arg to parameterise by reach.

**Reply by tamvejarf** [reply]
I followed your advice, but I encountered unusual .tpl outputs, as shown in the image. I'm not sure if I'm configuring the parameters incorrectly (which is likely) or if it would be more practical to generate a vector for the conductance, use the add_parameters function with that vector, and then create another Python script to rebuild the original package.

![riv2](https://github.com/user-attachments/assets/9338b192-0235-4436-8d42-9cbb4801f017)

**Reply by briochh** [reply]
This looks about as expected. The tpl files are slightly unorthodox, they work with the pyemu apply function that is added to the forward_run.py during the PstFrom.build_pst() step. It can be a little confusing to those used to the classic tpl mirroring of model input files. These ones actually mirror files that get placed in the mult directory.
